cmd/deployctl/cmd/build.go

```diff
@@ -330,6 +330,9 @@ func CollectCmdRun(cmd *cobra.Command, args []string) {
 		os.Exit(40)
 	}
 
+	//Remove controller0Address and controller0Address if the values are empty
+	build.CleanEmptyAttributes(deployment)
```

Comment:
The method should not be placed in the top level. Suggest to move into GetAssociatedAddressPools

Reply:
updated as filter